src/nile/cli.py

```diff
@@ -171,16 +172,18 @@ async def declare(
     alias,
     overriding_path,
     token,
+    nile_account,
 ):
     """Declare a StarkNet smart contract through an Account."""
-    account = await Account(signer, network)
+    account = await Account(signer, network, watch_mode=watch_mode)
     await account.declare(
         contract_name,
         alias=alias,
         max_fee=max_fee,
         overriding_path=overriding_path,
         mainnet_token=token,
         watch_mode=watch_mode,
+        nile_account=nile_account,
```

Comment:
what about making it a constructor argument and object attribute? i.e. `self.is_nile_account`

it could be leaner if we eventually need it for other operations, and it would debloat the function's arguments. although right now it's the same since it's just here

Reply:
We can maybe add this in the future, but I don't think this would work here. The point of `nile_account` is to direct the path to the Nile artifacts. Really, it's just for declaring a Nile account, so we need a way to adjust the path for this specific type of declaration. In other words, when an account wants to declare a Nile account, use the flag to change the path. For other declarations, Nile assumes the local `artifacts/` path.

IMO a better fix would be to iterate both paths checking for the target contract. That way, we don't have to worry about any of this. I already have a working example of this improvement, but for the sake of getting this shipped, I decided not to include it